src/components/Project/Secrets/paginator.js

```diff
@@ -351,7 +362,8 @@ export default class SecretsPaginator extends React.Component {
     if(!force && this.form.isDirty){
       this.setState({ dirtyFormDialogOpen: true })
     } else {
-      this.setState({ drawerOpen: false, addEnvVarMenuOpen: false, saving: true, dialogOpen: false, dirtyFormDialogOpen: false })
+      this.form = this.initProjectSecretsForm()
+      this.setState({ selectedSecret: null, selectedSecretID: null, drawerOpen: false, addEnvVarMenuOpen: false, saving: true, dialogOpen: false, dirtyFormDialogOpen: false })
```

Comment:
onError sets saving to true?

Reply:
Also, since this is setting a lot of params can we break it out so it's not a one-liner for readability?
That is to say can you please format it 'struct' style.